cmd/handlers/main.go

```diff
@@ -136,7 +149,15 @@ func main() {
 			} else {
 
 				// Add block to blockchain
-				if err := blockchain.AddBlock(newBlock, constants.BlockchainFile, constants.MetadataTable); err != nil {
+				blockchainFile, err := os.OpenFile(constants.BlockchainFile, os.O_APPEND|os.O_WRONLY, 0644)
+				if err != nil {
+					log.Fatalf("failed to open ledger file: %s\n", err)
+				}
+				err = blockchain.AddBlock(newBlock, blockchainFile, metadataDatabaseConnection)
+				if errClosing := blockchainFile.Close(); errClosing != nil {
+					log.Fatalf("Failed to close blockchain file: %v", err)
```

Comment:
Why is `errClosing` being declared if we print out `err` in the `log.Fatalf`?

Reply:
Oh, `log.Fatalf` should print `errClosing` rather than `err` here.

Reply:
Instead of `errClosing` you can just call it `err`. It will be a different `err` than the one on line 156, because its within its own scope. That one should also be handled though.

- [ ] handle `AddBlock` `err`
- [ ] use `err` instead of `errClosing` in `if` block

Reply:
Shouldn't we handle the `AddBlock` error first? It seems if it is not nil, we log fatally anyway. If it is nil, then in the else block starting on line 162 we can then close it, and handle that error.

- [ ] Handle `AddBlock` error first; if it works correctly, handle the `Close` error